site/pages/create-social-media-post/[disruptionId]/[socialMediaPostIndex].page.tsx

```diff
@@ -0,0 +1,200 @@
+import { NextPageContext } from "next";
+import { parseCookies } from "nookies";
+import { ReactElement, useState } from "react";
+import ErrorSummary from "../../../components/ErrorSummary";
+import CsrfForm from "../../../components/form/CsrfForm";
+import DateSelector from "../../../components/form/DateSelector";
+import FormElementWrapper, { FormGroupWrapper } from "../../../components/form/FormElementWrapper";
+import Select from "../../../components/form/Select";
+import TextInput from "../../../components/form/TextInput";
+import TimeSelector from "../../../components/form/TimeSelector";
+import { BaseLayout } from "../../../components/layout/Layout";
+import { COOKIES_SOCIAL_MEDIA_ERRORS } from "../../../constants";
+import { getDisruptionById } from "../../../data/dynamo";
+import { PageState } from "../../../interfaces";
+import { SocialMediaPost, socialMediaPostSchema } from "../../../schemas/social-media.schema";
+import { destroyCookieOnResponseObject, getPageState } from "../../../utils/apiUtils";
+import { getSession } from "../../../utils/apiUtils/auth";
+import { getStateUpdater } from "../../../utils/formUtils";
+
+const title = "Create social media message";
+const description = "Create social media message page for the Create Transport Disruptions Service";
+
+export interface CreateSocialMediaPostPageProps extends PageState<Partial<SocialMediaPost>> {
+    disruptionSummary: string;
+    socialMediaPostIndex: number;
+    csrfToken?: string;
+}
+
+const CreateSocialMediaPost = (props: CreateSocialMediaPostPageProps): ReactElement => {
+    const [pageState, setPageState] = useState<PageState<Partial<SocialMediaPost>>>(props);
+
+    const stateUpdater = getStateUpdater(setPageState, pageState);
+    return (
+        <BaseLayout title={title} description={description}>
+            <CsrfForm
+                encType="multipart/form-data"
+                action={`/api/create-social-media-post?_csrf=${props.csrfToken || ""}`}
+                method="post"
+                csrfToken={props.csrfToken}
+                hideCsrf
```

Comment:
I'm thinking rather than modify the CsrfForm here, we could just use the standard form element:

```tsx
<form
    encType="multipart/form-data"
    action={`/api/create-social-media-post?_csrf=${props.csrfToken || ""}`}
    method="post"
/>
```

Reply:
done